packages/devextreme/js/__internal/grids/grid_core/m_modules.ts

```diff
@@ -256,6 +256,36 @@ const View: ModuleType<ViewType> = ModuleItem.inherit({
     return this.component._getTemplate(name);
   },
 
+  getView(name) {
+    return this.component._views?.[name];
+  },
+
+  getFirstVisibleView(viewNames, checkLastFirst = false) {
+    checkLastFirst && viewNames.reverse();
+    return viewNames.find((name) => this.getView(name)?.isVisible?.());
+  },
+
+  updateBorderedViews() {
+    const BORDERED_TOP_VIEW_CLASS = 'dx-bordered-top-view';
+    const BORDERED_BOTTOM_VIEW_CLASS = 'dx-bordered-bottom-view';
+
+    const BORDERED_TOP_VIEW_NAMES = ['columnHeadersView', 'rowsView'];
+    const BORDERED_BOTTOM_VIEW_NAMES = ['rowsView', 'footerView', 'filterPanelView'];
+
+    const firstVisibleTopViewName = this.getFirstVisibleView(BORDERED_TOP_VIEW_NAMES);
+    const lastVisibleBottomViewName = this.getFirstVisibleView(BORDERED_BOTTOM_VIEW_NAMES, true);
+
+    each(BORDERED_TOP_VIEW_NAMES, (index, viewName) => {
```

Comment:
Why use each function, why not use the standard array iteration?

Reply:
No specific reason actually. In any case, I need to refactor again. So, I'll use standard methods instead if needed.